src/puppetlabs/trapperkeeper/services/authorization/authorization_service.clj

```diff
@@ -20,6 +21,9 @@
           (assoc-in [:allow-header-cert-info] (get config
                                                    :allow-header-cert-info
                                                    false)))))
+  (authorization-check [this request]
+   (let [{:keys [rules allow-header-cert-info]} (service-context this)]
+    (ring-middleware/authorization-check request rules allow-header-cert-info)))
```

Comment:
I think it would be good for calls to this function to have query-params destructured into the request - like is done for calls to `wrap-with-authorization-check` via the `wrap-query-params` middleware - before the request is authorized.  This would be useful in the event that the request needs to be validated against rules that might have `query-params` constraints on them.

Now that we'd have a couple of paths into the core middleware functions, I wonder if it would be better to just call `assoc-query-params` on the request - like the `ring-middleware/wrap-query-params` currently does - from inside of the new `ring-middleware/authorization-check` function.  This could be done to the request right before it is threaded through to the `rules/allowed?` call - and retained on the request that is returned in the response map from `ring-middleware/authorization-check`.

@rlinehan, @nwolfe, @dankreek - WDYT?